x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -174,6 +176,25 @@ protected static List<Batch<LogicalPlan>> rules() {
         return asList(substitutions, operators(), skip, cleanup(), defaultTopN, label);
     }
 
+    /**
+     * Currently this works similarly to SurrogateExpression, leaving the logic inside the expressions,
+     * so each can decide for itself whether or not to change to a surrogate expression.
+     * But what is actually being done is similar to LiteralsOnTheRight. We can consider in the future moving
+     * this in either direction, reducing the number of rules, but for now,
+     * it's a separate rule to reduce the risk of unintended interactions with other rules.
+     */
+    static class SubstituteSpatialSurrogates extends OptimizerRules.OptimizerExpressionRule<SpatialRelatesFunction> {
```

Comment:
Move it under SubstituteSurrogate and please raise an issue to indicate that this needs to either be updated based on either LiteralsOnTheRight OR be incorporated into SubstituteSurrogates

Reply:
Moved and created issue at https://github.com/elastic/elasticsearch/issues/106763